cli/src/commands/parse/pst.rs

```diff
@@ -0,0 +1,1275 @@
+use anyhow::{anyhow, Context, Result};
+use chrono::{DateTime, TimeZone, Utc};
+use colored::Colorize;
+use itertools::Itertools;
+use libc::{c_char, c_int, size_t};
+use mailparse::{parse_header, MailHeader, MailHeaderMap};
+use reinfer_client::{
+    resources::{
+        attachments::AttachmentMetadata,
+        email::{EmailMetadata, Id as EmailId},
+    },
+    BucketIdentifier, Client, Mailbox, MimeContent, NewEmail,
+};
+use std::{
+    collections::HashSet,
+    path::{Path, PathBuf},
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc,
+    },
+};
+use structopt::StructOpt;
+
+use crate::progress::{Options as ProgressOptions, Progress};
+
+const ROOT_FOLDER_NAME: &str = "root";
+
+#[repr(C)]
+#[derive(Debug, Clone)]
+struct LibPffItem {
+    _unused: c_int,
+}
+
+#[repr(C)]
+#[derive(Debug, Clone)]
+struct LibPffRecordEntry {
+    _unused: c_int,
+}
+
+#[repr(C)]
+#[derive(Debug, Clone)]
+struct LibPffRecordSet {
+    _unused: c_int,
+}
+
+#[repr(C)]
+#[derive(Debug, Clone)]
+struct LibPffFile {
+    _unused: c_int,
+}
+
+#[repr(C)]
+#[derive(Debug, Clone)]
+struct LibcError {
+    number_of_messages: c_int,
+    messages: *const *const c_char,
+}
+
+#[derive(Clone)]
+enum LibPffEntryType {
+    MessageTransportHeaders = 0x007d,
+    AttachmentFilenameLong = 0x3707,
+}
+
+#[derive(Clone)]
+enum LibPffEntryValueFlags {
+    MatchAnyValueType = 0x01,
+}
+
+impl LibcError {
+    pub fn as_string(&self) -> String {
+        let mut errors: Vec<String> = Vec::new();
+        for i in 0..self.number_of_messages {
+            let message_ptr = unsafe { self.messages.offset(i as isize) };
+
+            if message_ptr.is_null() {
+                errors.push("Unknown Error: message_ptr null".to_string());
+                continue;
+            }
+
+            if !message_ptr.is_aligned() {
+                errors.push("Unknown Error: message_ptr misaligned".to_string());
+                continue;
+            }
+
+            let message = unsafe { std::ffi::CStr::from_ptr(*message_ptr) }
+                .to_string_lossy()
+                .into_owned();
+            errors.push(message);
+        }
+
+        // Return errors deduplicated
+        errors
+            .into_iter()
+            .collect::<HashSet<String>>()
+            .iter()
+            .join("\n")
+    }
+}
+
+#[link(name = "pff")]
+extern "C" {
```

Comment:
The low level C wrapper should/can be moved out to a separate module called something like `pff-sys`. Basically keep all the low level C-repr details in another file, which basically defines the `libpff` C interface

See this thread: https://users.rust-lang.org/t/should-i-follow-the-sys-crate-convention-for-wrapping-non-system-c-libraries/65150